cpp/src/io/parquet/parquet.hpp

```diff
@@ -183,6 +183,9 @@ struct SchemaElement {
   // extra cudf specific fields
   bool output_as_byte_array = false;
 
+  // cudf type determined from arrow:schema
+  thrust::optional<type_id> arrow_type;
```

Comment:
This is only used for duration types currently, correct? Maybe change the name to reflect that (see below in reader_impl_helpers)

Reply:
Although I don't mind updating the name I am partial to keeping the name generic here like `arrow_type` instead of specific to duration as there certainly are other arrow types for which we might require this in (near) future.

Reply:
Fair enough.  All the more reason then to exit early from `to_type_id()` if the `type_id` has already been determined via the arrow schema. No need to mix that with the physical type handling.

Reply:
All done!